src/biogeophys/BiogeophysPreFluxCalcsMod.F90

```diff
@@ -200,16 +200,11 @@ subroutine SetZ0mDisp(bounds, num_nolakep, filter_nolakep, &
                          / 2._r8)**(-0.5_r8) /  (pftcon%z0v_LAImax(patch%itype(p))) / pftcon%z0v_c(patch%itype(p))
 
                if ( htop(p) <= 1.e-10_r8 )then
-                  if (lun%itype(l) == istcrop) then
-                     z0m(p) = 0._r8
-                     displa(p) = 0._r8
-                  else
-                     write(iulog,*) ' nstep = ', get_nstep(), ' htop = ', htop(p)
-                     call endrun(subgrid_index=p, subgrid_level=subgrid_level_patch, msg=errMsg(sourcefile, __LINE__))
-                  end if
+                  z0m(p) = 0._r8
+                  displa(p) = 0._r8
```

Comment:
In case it helps to look at the origins of this code, see 
BiogeophysPreFluxCalcsMod.F90
in
https://github.com/ESCOMP/CTSM/pull/2045/files

The if statement came in with the Meier2022 code to prevent divide by zero (or tiny) htop. The if statement did not account for the possibility of valid zero (or tiny) htop.